content/en/docs/hertz/tutorials/advanced-exten/protocol.md

```diff
@@ -0,0 +1,166 @@
+
+In this logic, we can customize the protocol parsing method, introduce business Handler execution, data write back and other standard behaviors of the protocol layer. This is also the core logic of our custom server.
+
+```go
+type myServer struct{
+    xxx
+    xxx
+}
+
+func (s *myServer) Serve (c context.Context, conn network.Conn) error{
+    // protocol parsing
+	...
+    // Go to the logic function of business registration (Route, Middleware, Handler...)
+	...
+    // write data back
+	...
+}
+```
+
+Defining a protocol processing logic is as simple as that! However, the two steps of parsing the protocol and writing data back can be easily achieved through the `conn` interface provided in the input parameters, but how to go to the logical function of business registration?
+
+### How to interact with upper-level logic
+
+A complete protocol must introduce business logic control (except for very few special situations), so how does the custom protocol in the Hertz framework realize this part of the ability? In fact, in the process of custom server initialization, the framework has naturally handed over this part of the capabilities to the custom protocol server.
+
+```go
+type ServerFactory interface {
+   New(core Core) (server protocol.Server, err error)
+}
+
+// Core is the core interface that promises to be provided for the protocol layer extensions
+type Core interface {
+   // IsRunning Check whether engine is running or not
+   IsRunning() bool
+   // A RequestContext pool ready for protocol server impl
+   GetCtxPool() *sync.Pool
+   // Business logic entrance
+   // After pre-read works, protocol server may call this method
+   // to introduce the middlewares and handlers
+   ServeHTTP(c context.Context, ctx *app.RequestContext)
+   // GetTracer for tracing requirement
+   GetTracer() tracer.Controller
+}
+```
+
+A custom server only needs to implement a protocol server generation factory according to the above interface. The Core in the parameters actually includes the introduction of upper-layer logic interaction and the specific implementation of other core application layer interfaces. When initializing a custom server, normally you only need to save the Core to the server. When you need to transfer to the business logic, you can guide the process to the application layer processing logic (Route, Middleware, Logic Handler) through the Core. When the business logic is executed and returned, you can further packet writeback is performed based on the business data.
+
+```go
+type myServer struct{
+    suite.Core
+    xxx
+}
+
+func (s *myServer) Serve (c context.Context, conn network.Conn) error{
+    // protocol parsing
+	...
+    Core.ServeHTTP(c, ctx)
+    // write data back
+	...
+}
+```
+
+So far, a custom protocol layer server has been developed.
+
+### Register custom protocol server with Hertz
```

Comment:
Registration of custom protocol server into Hertz

-> all those titles use noun form to be consistent